src/NovelWorldObject.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+#include "NovelWorldObject.h"
+#include "GeoBounds.h"
+
+namespace NovelRT {
+NovelWorldObject::NovelWorldObject(NovelLayeringService* layeringService,
+                                   const NovelCommonArgs& args) :
+    _layeringService(layeringService) {
+  setLayer(args.layer);
+  setOrderInLayer(args.orderInLayer);
+  _layeringService->updateLayerInfo(getLayer(), this, true);
+  setPosition(args.startingPosition);
+  setRotation(args.startingRotation);
+  setScale(args.startingScale);
+  setActive(true);
+}
+
+NovelWorldObject::~NovelWorldObject() = default;
+
+GeoVector<float> NovelWorldObject::getPosition() const {
```

Comment:
several of these should probably be in the header file so they can be inlined/etc

Reply:
Discussed in Discord, can't do this due to method overrides.